### PR TITLE
Add Dialog::run_future()

### PR DESCRIPTION
**Pull request by sophie-h**
Backport of gtk-rs/gtk4-rs#149

### CODE REVIEW

**Review by sdroege — APPROVED** [review]

File: examples/Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,6 @@
 name = "gtk-rs-examples"
 version = "0.0.1"
 authors = ["The Gtk-rs Project Developers"]
-autobins = false
```
Comment (GuillaumeGomez) [inline examples/Cargo.toml:5]: Why this change?

Comment (sophie-h) [reply examples/Cargo.toml:5]: Otherwise I would have to add a `[[bin]]` section which would clash with #267

Comment (GuillaumeGomez) [reply examples/Cargo.toml:5]: I merged #267 so maybe rebase? ;)

Comment (sophie-h) [reply examples/Cargo.toml]: Thanks! Done
